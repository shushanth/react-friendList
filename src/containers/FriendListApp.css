body {
  align-items: center;
  background-color: #f4f3f0;
  display: flex;
  justify-content: center;
}

* {
  padding: 0;
  margin: 0;
}

:local(.friendListApp) {
  background-color: #5c75b0;
  border: 1px solid #e3e3e3;
  padding-top: 18px;
  width: 300px;
}

:local(.friendListApp h1) {
  color: white;
  font-size: 16px;
  font-family: Helvetica;
  line-height: 20px;
  margin-bottom: 10px;
  margin-top: 0;
  padding-left: 10px;
}
