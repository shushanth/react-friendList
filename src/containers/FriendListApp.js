import React from "react";
import styles from "./FriendListApp.css";
import { connect } from "react-redux";

import {
  addFriend,
  deleteFriend,
  starFriend,
  pageChange,
  pageMove
} from "../actions/friendsActions";
import { FriendList, AddFriendForm, FriendPagination } from "../components";

const FriendListApp = props => {
  const {
    friendlist: { friendsById, selectedPage, pageSize },
    addFriend,
    deleteFriend,
    starFriend,
    pageChange,
    pageMove
  } = props;

  const actions = {
    addFriend,
    deleteFriend,
    starFriend,
    pageChange,
    pageMove
  };

  let totalPages = friendsById.length || 0;
  return (
    <div className={styles.friendListApp}>
      <h1>The FriendList</h1>
      <AddFriendForm
        pageMove={actions.pageMove}
        addFriend={actions.addFriend}
      />
      <FriendList
        friends={friendsById}
        actions={actions}
        selectedPage={selectedPage}
        pageSize={pageSize}
      />
      {!(totalPages <= pageSize) ? (
        <FriendPagination
          selectedPage={selectedPage}
          pageSize={pageSize}
          totalPages={totalPages}
          onPageChange={actions.pageChange}
        />
      ) : null}
    </div>
  );
};

function mapStateToProps(state) {
  return state;
}

export default connect(
  mapStateToProps,
  {
    addFriend,
    deleteFriend,
    starFriend,
    pageChange,
    pageMove
  }
)(FriendListApp);
