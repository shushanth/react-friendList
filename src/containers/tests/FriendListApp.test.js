import React from "react";
import { shallow } from "enzyme";

import FriendListApp from "../FriendListApp";

//as the component is pure component
const FriendListAppComponent = FriendListApp.WrappedComponent;

describe("Component: <FriendListApp/>", () => {
  let FriendListAppWrapper, props;

  beforeAll(() => {
    props = {
      friendlist: {
        ...{
          selectedPage: 1,
          pageSize: 2,
          friendsById: [
            {
              id: 1,
              name: "Theodore Roosevelt",
              gender: "Male",
              starred: true
            }
          ]
        }
      },
      addFriend: jest.fn(),
      deleteFriend: jest.fn(),
      starFriend: jest.fn(),
      pageChange: jest.fn(),
      pageMove: jest.fn()
    };
    FriendListAppWrapper = shallow(<FriendListAppComponent {...props} />);
  });

  it("should render without crashing", () => {
    expect(FriendListAppWrapper.exists()).toBe(true);
    expect(FriendListAppWrapper).toMatchSnapshot();
  });

  it("should have heading title in h1", () => {
    let heading = FriendListAppWrapper.find("h1").text();
    expect(heading).toEqual("The FriendList");
  });

  it("should have add friend component", () => {
    let addFriendComponent = FriendListAppWrapper.find("AddFriendForm");
    expect(addFriendComponent.exists()).toBe(true);
    expect(addFriendComponent).toMatchSnapshot();
  });

  it("should have friend list component", () => {
    let friendListComponent = FriendListAppWrapper.find("FriendList");
    expect(friendListComponent.exists()).toBe(true);
    expect(friendListComponent).toMatchSnapshot();
  });

  it("should have not friend pagination component, when friends are less than two", () => {
    let friendPaginationComponent = FriendListAppWrapper.find(
      "FriendPagination"
    );
    expect(friendPaginationComponent).toMatchSnapshot();
  });
});
