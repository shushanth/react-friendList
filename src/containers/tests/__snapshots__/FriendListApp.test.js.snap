exports[`Component: <FriendListApp/> should have add friend component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <FriendListApp
      addFriend={[Function]}
      deleteFriend={[Function]}
      friendlist={
            Object {
                  "friendsById": Array [
                    Object {
                      "gender": "Male",
                      "id": 1,
                      "name": "Theodore Roosevelt",
                      "starred": true,
                    },
                  ],
                  "pageSize": 2,
                  "selectedPage": 1,
                }
      }
      pageChange={[Function]}
      pageMove={[Function]}
      starFriend={[Function]} />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            The FriendList
</h1>,
          <AddFriendForm
            addFriend={[Function]}
            pageMove={[Function]} />,
          <FriendList
            actions={
                        Object {
                                    "addFriend": [Function],
                                    "deleteFriend": [Function],
                                    "pageChange": [Function],
                                    "pageMove": [Function],
                                    "starFriend": [Function],
                                  }
            }
            friends={
                        Array [
                                    Object {
                                      "gender": "Male",
                                      "id": 1,
                                      "name": "Theodore Roosevelt",
                                      "starred": true,
                                    },
                                  ]
            }
            pageSize={2}
            selectedPage={1} />,
          null,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The FriendList",
          },
          "ref": null,
          "rendered": "The FriendList",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addFriend": [Function],
            "pageMove": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "actions": Object {
              "addFriend": [Function],
              "deleteFriend": [Function],
              "pageChange": [Function],
              "pageMove": [Function],
              "starFriend": [Function],
            },
            "friends": Array [
              Object {
                "gender": "Male",
                "id": 1,
                "name": "Theodore Roosevelt",
                "starred": true,
              },
            ],
            "pageSize": 2,
            "selectedPage": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              The FriendList
</h1>,
            <AddFriendForm
              addFriend={[Function]}
              pageMove={[Function]} />,
            <FriendList
              actions={
                            Object {
                                          "addFriend": [Function],
                                          "deleteFriend": [Function],
                                          "pageChange": [Function],
                                          "pageMove": [Function],
                                          "starFriend": [Function],
                                        }
              }
              friends={
                            Array [
                                          Object {
                                            "gender": "Male",
                                            "id": 1,
                                            "name": "Theodore Roosevelt",
                                            "starred": true,
                                          },
                                        ]
              }
              pageSize={2}
              selectedPage={1} />,
            null,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "The FriendList",
            },
            "ref": null,
            "rendered": "The FriendList",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addFriend": [Function],
              "pageMove": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "actions": Object {
                "addFriend": [Function],
                "deleteFriend": [Function],
                "pageChange": [Function],
                "pageMove": [Function],
                "starFriend": [Function],
              },
              "friends": Array [
                Object {
                  "gender": "Male",
                  "id": 1,
                  "name": "Theodore Roosevelt",
                  "starred": true,
                },
              ],
              "pageSize": 2,
              "selectedPage": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          null,
        ],
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactFifteenAdapter {
        "options": Object {
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "prevContext": true,
            },
          },
          "supportPrevContextArgumentOfComponentDidUpdate": true,
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "addFriend": [Function],
      "pageMove": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "addFriend": [Function],
        "pageMove": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            The FriendList
</h1>,
          <AddFriendForm
            addFriend={[Function]}
            pageMove={[Function]} />,
          <FriendList
            actions={
                        Object {
                                    "addFriend": [Function],
                                    "deleteFriend": [Function],
                                    "pageChange": [Function],
                                    "pageMove": [Function],
                                    "starFriend": [Function],
                                  }
            }
            friends={
                        Array [
                                    Object {
                                      "gender": "Male",
                                      "id": 1,
                                      "name": "Theodore Roosevelt",
                                      "starred": true,
                                    },
                                  ]
            }
            pageSize={2}
            selectedPage={1} />,
          null,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The FriendList",
          },
          "ref": null,
          "rendered": "The FriendList",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addFriend": [Function],
            "pageMove": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "actions": Object {
              "addFriend": [Function],
              "deleteFriend": [Function],
              "pageChange": [Function],
              "pageMove": [Function],
              "starFriend": [Function],
            },
            "friends": Array [
              Object {
                "gender": "Male",
                "id": 1,
                "name": "Theodore Roosevelt",
                "starred": true,
              },
            ],
            "pageSize": 2,
            "selectedPage": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": "div",
    },
  ],
}
`;

exports[`Component: <FriendListApp/> should have friend list component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <FriendListApp
      addFriend={[Function]}
      deleteFriend={[Function]}
      friendlist={
            Object {
                  "friendsById": Array [
                    Object {
                      "gender": "Male",
                      "id": 1,
                      "name": "Theodore Roosevelt",
                      "starred": true,
                    },
                  ],
                  "pageSize": 2,
                  "selectedPage": 1,
                }
      }
      pageChange={[Function]}
      pageMove={[Function]}
      starFriend={[Function]} />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            The FriendList
</h1>,
          <AddFriendForm
            addFriend={[Function]}
            pageMove={[Function]} />,
          <FriendList
            actions={
                        Object {
                                    "addFriend": [Function],
                                    "deleteFriend": [Function],
                                    "pageChange": [Function],
                                    "pageMove": [Function],
                                    "starFriend": [Function],
                                  }
            }
            friends={
                        Array [
                                    Object {
                                      "gender": "Male",
                                      "id": 1,
                                      "name": "Theodore Roosevelt",
                                      "starred": true,
                                    },
                                  ]
            }
            pageSize={2}
            selectedPage={1} />,
          null,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The FriendList",
          },
          "ref": null,
          "rendered": "The FriendList",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addFriend": [Function],
            "pageMove": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "actions": Object {
              "addFriend": [Function],
              "deleteFriend": [Function],
              "pageChange": [Function],
              "pageMove": [Function],
              "starFriend": [Function],
            },
            "friends": Array [
              Object {
                "gender": "Male",
                "id": 1,
                "name": "Theodore Roosevelt",
                "starred": true,
              },
            ],
            "pageSize": 2,
            "selectedPage": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              The FriendList
</h1>,
            <AddFriendForm
              addFriend={[Function]}
              pageMove={[Function]} />,
            <FriendList
              actions={
                            Object {
                                          "addFriend": [Function],
                                          "deleteFriend": [Function],
                                          "pageChange": [Function],
                                          "pageMove": [Function],
                                          "starFriend": [Function],
                                        }
              }
              friends={
                            Array [
                                          Object {
                                            "gender": "Male",
                                            "id": 1,
                                            "name": "Theodore Roosevelt",
                                            "starred": true,
                                          },
                                        ]
              }
              pageSize={2}
              selectedPage={1} />,
            null,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "The FriendList",
            },
            "ref": null,
            "rendered": "The FriendList",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addFriend": [Function],
              "pageMove": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "actions": Object {
                "addFriend": [Function],
                "deleteFriend": [Function],
                "pageChange": [Function],
                "pageMove": [Function],
                "starFriend": [Function],
              },
              "friends": Array [
                Object {
                  "gender": "Male",
                  "id": 1,
                  "name": "Theodore Roosevelt",
                  "starred": true,
                },
              ],
              "pageSize": 2,
              "selectedPage": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          null,
        ],
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactFifteenAdapter {
        "options": Object {
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "prevContext": true,
            },
          },
          "supportPrevContextArgumentOfComponentDidUpdate": true,
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "actions": Object {
        "addFriend": [Function],
        "deleteFriend": [Function],
        "pageChange": [Function],
        "pageMove": [Function],
        "starFriend": [Function],
      },
      "friends": Array [
        Object {
          "gender": "Male",
          "id": 1,
          "name": "Theodore Roosevelt",
          "starred": true,
        },
      ],
      "pageSize": 2,
      "selectedPage": 1,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "actions": Object {
          "addFriend": [Function],
          "deleteFriend": [Function],
          "pageChange": [Function],
          "pageMove": [Function],
          "starFriend": [Function],
        },
        "friends": Array [
          Object {
            "gender": "Male",
            "id": 1,
            "name": "Theodore Roosevelt",
            "starred": true,
          },
        ],
        "pageSize": 2,
        "selectedPage": 1,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            The FriendList
</h1>,
          <AddFriendForm
            addFriend={[Function]}
            pageMove={[Function]} />,
          <FriendList
            actions={
                        Object {
                                    "addFriend": [Function],
                                    "deleteFriend": [Function],
                                    "pageChange": [Function],
                                    "pageMove": [Function],
                                    "starFriend": [Function],
                                  }
            }
            friends={
                        Array [
                                    Object {
                                      "gender": "Male",
                                      "id": 1,
                                      "name": "Theodore Roosevelt",
                                      "starred": true,
                                    },
                                  ]
            }
            pageSize={2}
            selectedPage={1} />,
          null,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The FriendList",
          },
          "ref": null,
          "rendered": "The FriendList",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addFriend": [Function],
            "pageMove": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "actions": Object {
              "addFriend": [Function],
              "deleteFriend": [Function],
              "pageChange": [Function],
              "pageMove": [Function],
              "starFriend": [Function],
            },
            "friends": Array [
              Object {
                "gender": "Male",
                "id": 1,
                "name": "Theodore Roosevelt",
                "starred": true,
              },
            ],
            "pageSize": 2,
            "selectedPage": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": "div",
    },
  ],
}
`;

exports[`Component: <FriendListApp/> should have not friend pagination component, when friends are less than two 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <FriendListApp
      addFriend={[Function]}
      deleteFriend={[Function]}
      friendlist={
            Object {
                  "friendsById": Array [
                    Object {
                      "gender": "Male",
                      "id": 1,
                      "name": "Theodore Roosevelt",
                      "starred": true,
                    },
                  ],
                  "pageSize": 2,
                  "selectedPage": 1,
                }
      }
      pageChange={[Function]}
      pageMove={[Function]}
      starFriend={[Function]} />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            The FriendList
</h1>,
          <AddFriendForm
            addFriend={[Function]}
            pageMove={[Function]} />,
          <FriendList
            actions={
                        Object {
                                    "addFriend": [Function],
                                    "deleteFriend": [Function],
                                    "pageChange": [Function],
                                    "pageMove": [Function],
                                    "starFriend": [Function],
                                  }
            }
            friends={
                        Array [
                                    Object {
                                      "gender": "Male",
                                      "id": 1,
                                      "name": "Theodore Roosevelt",
                                      "starred": true,
                                    },
                                  ]
            }
            pageSize={2}
            selectedPage={1} />,
          null,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The FriendList",
          },
          "ref": null,
          "rendered": "The FriendList",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addFriend": [Function],
            "pageMove": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "actions": Object {
              "addFriend": [Function],
              "deleteFriend": [Function],
              "pageChange": [Function],
              "pageMove": [Function],
              "starFriend": [Function],
            },
            "friends": Array [
              Object {
                "gender": "Male",
                "id": 1,
                "name": "Theodore Roosevelt",
                "starred": true,
              },
            ],
            "pageSize": 2,
            "selectedPage": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              The FriendList
</h1>,
            <AddFriendForm
              addFriend={[Function]}
              pageMove={[Function]} />,
            <FriendList
              actions={
                            Object {
                                          "addFriend": [Function],
                                          "deleteFriend": [Function],
                                          "pageChange": [Function],
                                          "pageMove": [Function],
                                          "starFriend": [Function],
                                        }
              }
              friends={
                            Array [
                                          Object {
                                            "gender": "Male",
                                            "id": 1,
                                            "name": "Theodore Roosevelt",
                                            "starred": true,
                                          },
                                        ]
              }
              pageSize={2}
              selectedPage={1} />,
            null,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "The FriendList",
            },
            "ref": null,
            "rendered": "The FriendList",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addFriend": [Function],
              "pageMove": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "actions": Object {
                "addFriend": [Function],
                "deleteFriend": [Function],
                "pageChange": [Function],
                "pageMove": [Function],
                "starFriend": [Function],
              },
              "friends": Array [
                Object {
                  "gender": "Male",
                  "id": 1,
                  "name": "Theodore Roosevelt",
                  "starred": true,
                },
              ],
              "pageSize": 2,
              "selectedPage": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          null,
        ],
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactFifteenAdapter {
        "options": Object {
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "prevContext": true,
            },
          },
          "supportPrevContextArgumentOfComponentDidUpdate": true,
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            The FriendList
</h1>,
          <AddFriendForm
            addFriend={[Function]}
            pageMove={[Function]} />,
          <FriendList
            actions={
                        Object {
                                    "addFriend": [Function],
                                    "deleteFriend": [Function],
                                    "pageChange": [Function],
                                    "pageMove": [Function],
                                    "starFriend": [Function],
                                  }
            }
            friends={
                        Array [
                                    Object {
                                      "gender": "Male",
                                      "id": 1,
                                      "name": "Theodore Roosevelt",
                                      "starred": true,
                                    },
                                  ]
            }
            pageSize={2}
            selectedPage={1} />,
          null,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The FriendList",
          },
          "ref": null,
          "rendered": "The FriendList",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addFriend": [Function],
            "pageMove": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "actions": Object {
              "addFriend": [Function],
              "deleteFriend": [Function],
              "pageChange": [Function],
              "pageMove": [Function],
              "starFriend": [Function],
            },
            "friends": Array [
              Object {
                "gender": "Male",
                "id": 1,
                "name": "Theodore Roosevelt",
                "starred": true,
              },
            ],
            "pageSize": 2,
            "selectedPage": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": "div",
    },
  ],
}
`;

exports[`Component: <FriendListApp/> should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FriendListApp
    addFriend={[Function]}
    deleteFriend={[Function]}
    friendlist={
        Object {
            "friendsById": Array [
              Object {
                "gender": "Male",
                "id": 1,
                "name": "Theodore Roosevelt",
                "starred": true,
              },
            ],
            "pageSize": 2,
            "selectedPage": 1,
          }
    }
    pageChange={[Function]}
    pageMove={[Function]}
    starFriend={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          The FriendList
</h1>,
        <AddFriendForm
          addFriend={[Function]}
          pageMove={[Function]} />,
        <FriendList
          actions={
                    Object {
                              "addFriend": [Function],
                              "deleteFriend": [Function],
                              "pageChange": [Function],
                              "pageMove": [Function],
                              "starFriend": [Function],
                            }
          }
          friends={
                    Array [
                              Object {
                                "gender": "Male",
                                "id": 1,
                                "name": "Theodore Roosevelt",
                                "starred": true,
                              },
                            ]
          }
          pageSize={2}
          selectedPage={1} />,
        null,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The FriendList",
        },
        "ref": null,
        "rendered": "The FriendList",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "addFriend": [Function],
          "pageMove": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "actions": Object {
            "addFriend": [Function],
            "deleteFriend": [Function],
            "pageChange": [Function],
            "pageMove": [Function],
            "starFriend": [Function],
          },
          "friends": Array [
            Object {
              "gender": "Male",
              "id": 1,
              "name": "Theodore Roosevelt",
              "starred": true,
            },
          ],
          "pageSize": 2,
          "selectedPage": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            The FriendList
</h1>,
          <AddFriendForm
            addFriend={[Function]}
            pageMove={[Function]} />,
          <FriendList
            actions={
                        Object {
                                    "addFriend": [Function],
                                    "deleteFriend": [Function],
                                    "pageChange": [Function],
                                    "pageMove": [Function],
                                    "starFriend": [Function],
                                  }
            }
            friends={
                        Array [
                                    Object {
                                      "gender": "Male",
                                      "id": 1,
                                      "name": "Theodore Roosevelt",
                                      "starred": true,
                                    },
                                  ]
            }
            pageSize={2}
            selectedPage={1} />,
          null,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The FriendList",
          },
          "ref": null,
          "rendered": "The FriendList",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addFriend": [Function],
            "pageMove": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "actions": Object {
              "addFriend": [Function],
              "deleteFriend": [Function],
              "pageChange": [Function],
              "pageMove": [Function],
              "starFriend": [Function],
            },
            "friends": Array [
              Object {
                "gender": "Male",
                "id": 1,
                "name": "Theodore Roosevelt",
                "starred": true,
              },
            ],
            "pageSize": 2,
            "selectedPage": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
